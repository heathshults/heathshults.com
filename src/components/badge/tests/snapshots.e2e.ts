import { snapIt } from '../../../test';

const component = 'badge';

describe(component, () => {
  const snap = snapIt(component);

  describe('renders', () => {
    snap('renders correctly with no props', '<hs-badge>default</hs-badge>');
    snap('appends the color modifier class', '<hs-badge type="brand">brand</hs-badge>');
    snap('handles boolean rounded attribute', '<hs-badge rounded>default</hs-badge>');
    snap('handles both boolean rounded attribute and type', '<hs-badge rounded type="brand">brand</hs-badge>');
    snap('handles boolean ghost and boolean rounded', '<hs-badge ghost rounded>default</hs-badge>');
    snap('supports all the modifiers', '<hs-badge ghost rounded type="brand">brand</hs-badge>');
  });
});
