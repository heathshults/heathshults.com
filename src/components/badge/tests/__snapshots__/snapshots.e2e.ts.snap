// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`badge renders appends the color modifier class 1`] = `"<hs-badge type=\\"brand\\" class=\\"hydrated\\"><span class=\\"hs-badge hs-badge--brand\\">brand</span></hs-badge>"`;

exports[`badge renders handles boolean ghost and boolean rounded 1`] = `"<hs-badge ghost=\\"\\" rounded=\\"\\" class=\\"hydrated\\"><span class=\\"hs-badge hs-badge--rounded hs-badge--ghost\\">default</span></hs-badge>"`;

exports[`badge renders handles boolean rounded attribute 1`] = `"<hs-badge rounded=\\"\\" class=\\"hydrated\\"><span class=\\"hs-badge hs-badge--rounded\\">default</span></hs-badge>"`;

exports[`badge renders handles both boolean rounded attribute and type 1`] = `"<hs-badge rounded=\\"\\" type=\\"brand\\" class=\\"hydrated\\"><span class=\\"hs-badge hs-badge--brand hs-badge--rounded\\">brand</span></hs-badge>"`;

exports[`badge renders renders correctly with no props 1`] = `"<hs-badge class=\\"hydrated\\"><span class=\\"hs-badge\\">default</span></hs-badge>"`;

exports[`badge renders supports all the modifiers 1`] = `"<hs-badge ghost=\\"\\" rounded=\\"\\" type=\\"brand\\" class=\\"hydrated\\"><span class=\\"hs-badge hs-badge--brand hs-badge--rounded hs-badge--ghost\\">brand</span></hs-badge>"`;
